### ISSUE TITLE
DDEV not finding new commands

### ISSUE THREAD

**Issue by kreynen** [ddev/ddev-drupal-contrib#6]
I might be missing a step installing this, but when I run `ddev get ddev/ddev-drupal-contrib`, I get...

```
Downloading https://api.github.com/repos/ddev/ddev-drupal-contrib/tarball/1.0.0-rc11 
1.0.0-rc11_2362468473.tar.gz 74.05 KiB / ? [-----------------------------=----------------------------------] 758.25% 0s 

Installing project-level components: 
👍 commands 
👍 config.contrib.yaml 

Installed DDEV add-on ddev/ddev-drupal-contrib, use `ddev restart` to enable. 
```

I run ddev restart, but I when I try to run `ddev poser` or `ddev symlink-project`, the response is `Error: unknown command "poser" for "ddev" `

I'm running ddev version v1.21.6 if that matters.

**Comment by weitzman** [comment]
No idea why those commands would bve unavailable. Can you check if a command script is present at `.ddev/commands/web/poser`

**Comment by rfay** [comment]
You do have to ddev restart to activate new commands

**Comment by kreynen** [comment]
I am restarting. Commands for ddev-adminer don't appear to be added to .ddev/commands/host, but .global_commands/host.

Is the issue the lack of defining the commands in https://github.com/ddev/ddev-drupal-contrib/blob/main/install.yaml like https://github.com/ddev/ddev-adminer/blob/main/install.yaml#L9

I did find a similar error in https://github.com/ddev/ddev/issues/4253

**Comment by kreynen** [comment]
I'll also add that I'm running DDEV version v1.21.6 on MacOS 13.3.1 if that matters.

**Comment by tyler36** [comment]
I think @kreynen might have the wrong project type set..

Check your `.ddev/config.yaml` file and make sure you have the following set:

```yml
type: drupal10
docroot: web
php_version: "8.1"
```

The above is for Drupal 10; set `type` and `php_version` for older versions of Drupal.

The commands will only be available if `type` is set to `drupal8` , `drupal9` , `drupal10`.

## Background

My module has a `_` in its name, so I ran `ddev --project-name=my-module` which gave me the same errors because my config was set to:

```yml
name: my-module
type: php
docroot: ""
php_version: "8.1"
```

What I should have run was :
- `ddev config --project-name=my-module --project-type=drupal10 --docroot=web --php-version=8.1 `

... which gives me:

```yaml
name: my-module
type: drupal10
docroot: web
php_version: "8.1"
```

**Comment by weitzman** [comment]
Well spotted. Think I should allow php projects as well? Seems prudent.

**Comment by tyler36** [comment]
I think we should steer people to "correcting" the project type.
DDEV manages the `settings.ddev.local` file which contains database configuration etc. 

I drafted #7 for discussion.

Hopefully @kreynen will report back if this resolves their problem.

**Comment by kreynen** [comment]
@tyler36 thanks! 

My issue was that the project type was incorrect. So I didn't miss a step as much as using the DDEV defaults when following https://github.com/ddev/ddev-drupal-contrib/blob/main/README.md?plain=1#L6 results in not being able to run ddev poser. 

> I think we should steer people to "correcting" the project type.

Why not just give users the correct configuration to begin with?

After cloning a module in the current step 1, DDEV isn't going to detect this as a Drupal project. The user has to specify that. The web directory needed for the docroot doesn't exist after the clone step.

The instructions currently link to https://ddev.readthedocs.io/en/latest/users/project/, but neither the project README.txt nor the general DDEV instructions mention using the --project-name=my-module --project-type=drupal10 --docroot=web --php-version=8.1 options vs. what DDEV defaults to. When configuring DDEV using the prompts, setting the docroot to web before it exists results in a warning.

```
Docroot Location (current directory): web
Warning: the provided docroot at web does not currently exist.
```

When using the options @tyler36 immediately after cloning, I get `The provided docroot web does not exist. Allow ddev to create it with the --create-docroot flag. `

In https://github.com/ddev/ddev-drupal-contrib/pull/9 I suggest updating instructions in the README.md need to be updated to use `ddev config --project-name=my-module --project-type=drupal10 --docroot=web --create-docroot --php-version=8.1 `

Adding the DDEV config options was all I needed, but the Install steps made a number of assumptions (like already having DDEV installed). 

I don't think you need to add the information about editing the .ddev/config.yml in the initial steps. Anyone familiar with DDEV is going to know how to do that. All I needed was to understand that the ddev poser command was dependent on the DDEV config expecting a Drupal site BEFORE running ddev poser creates the structure that would allow DDEV to determine the config.

It's a little counter-intuitive, but now that I understand what's happening this is very easy to configure and makes contributing to module projects MUCH easier.

THANKS!

**Comment by tyler36** [comment]
> In https://github.com/ddev/ddev-drupal-contrib/pull/9 I suggest updating instructions in the README.md need to be updated to use ddev config --project-name=my-module --project-type=drupal10 --docroot=web --create-docroot --php-version=8.1

This is fine if you are using Drupal 10, but what if you on Drupal 9? Drupal 9 supports PHP 7.3 and 7.4, and 8.0 from Drupal 9.1.
This also needs changing when future versions of Drupal are released.

> `--create-docroot`

Good catch on this flag. Should definately include it in instructions somewhere.

>  Anyone familiar with DDEV is going to know how to do that. 

We can't assume people are familiar though. Providing clear instructions on how to do something and how to troubleshoot helps everyone, regardless of experience level.

**Comment by tyler36** [comment]
This issue can be closed because of #7 and #9 were merged.
